app/controllers/staff/reports_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class Staff::ReportsController < Staff::BaseController
+  include Secured
+
+  def show
+    @report = Report.find(id)
+    authorize @report
+
+    respond_to do |format|
+      format.html
+      format.csv do
+        response.headers["Content-Disposition"] = "attachment; filename=\"#{ERB::Util.url_encode(@report.description)}.csv\""
+      end
+    end
+  end
+
+  def edit
+    @report = Report.find(id)
+    authorize @report
+  end
+
+  def update
+    @report = Report.find(id)
+    authorize @report
+
+    @report.assign_attributes(report_params)
+    if @report.valid?
+      @report.save!
+      @report.create_activity key: "report.update", owner: current_user
+      activate_report
+      flash[:notice] = I18n.t("action.report.update.success")
+      redirect_to organisation_path(current_user.organisation)
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def id
+    params[:id]
+  end
+
+  def report_params
+    params.require(:report).permit(:deadline, :description)
+  end
+
+  def activate_report
```

Comment:
I'd maybe call this `activate!` as the controller is already for Reports. (The bang `!` because it's saving changes)

Reply:
Nice! I'll update this, thank you.